backend/src/connection.rs

```diff
@@ -38,8 +38,8 @@ const RECEIVER_REPORT_INTERVAL: Duration = Duration::from_secs(5);
 
 pub type PacketToSend = Vec<u8>;
 
-#[derive(Error, Debug, Eq, PartialEq)]
-pub enum Error {
+#[derive(PartialEq, Error, Debug, Eq)]
```

Comment:
This line probably doesn't need to change.

Reply:
Yeah, during the testing of test cases which are asserted against concreate type required the Error types to implement these traits! since `Option`  enum type inherently implements  `std::cmp` traits

Reply:
My comment was only asking why PartialEq	moved to the front of the list. We like to keep diffs as small as possible.

Reply:
Oh got it, Apologies I will refactor that 😊